iothub/device/src/DeviceAuthenticationWithX509Certificate.cs

```diff
@@ -19,10 +19,12 @@ public sealed class DeviceAuthenticationWithX509Certificate : IAuthenticationMet
         /// </summary>
         /// <param name="deviceId">Device Identifier.</param>
         /// <param name="certificate">X.509 Certificate.</param>
-        public DeviceAuthenticationWithX509Certificate(string deviceId, X509Certificate2 certificate)
+        /// <param name="chainCertificates">Certificates in the device certificate chain.</param>
+        public DeviceAuthenticationWithX509Certificate(string deviceId, X509Certificate2 certificate, X509Certificate2Collection chainCertificates = null)
```

Comment:
Another option is to add a new constructor. I chose to add it as it felt like less code and more discoverable. I'm open to suggestions.

Reply:
so: 
for self signed certs -> pass only certificate
for certs with a single parent -> pass only certificate = device cert, and upload the signing parent cert on hub
for certs with multiple certs in their hierarchy -> certificate = device cert, chainCertificates = chain from device cert to the point in the hierarchy whichever is upload to iot hub.

is this correct?

Reply:
hehe, looks like you've answered this right below!